src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -82,6 +82,29 @@
   private static final String BULK_SIZE_BYTES_DISPLAY = "Bulk Size (bytes)";
   private static final int BULK_SIZE_BYTES_DEFAULT = 5 * 1024 * 1024;
 
+  public static final String DATA_STREAM_DATASET_CONFIG = "data.stream.dataset";
+  private static final String DATA_STREAM_DATASET_DOC =
+      "Generic name describing data ingested and its structure to be written to data stream. "
+      + "Can be any arbitrary string that is no longer than 100 characters, is in all lowercase, "
+      + "and does not contain spaces or any of these special characters ``/\\*\"<>|,#:-``. "
+      + "Otherwise, the default is ``null`` which indicates the connector will write "
+      + "to regular indices instead. If filled, this configuration will be used alongside "
+      + "``data.stream.type`` to construct the index name.";
+  private static final String DATA_STREAM_DATASET_DISPLAY = "Data stream dataset";
+  private static final String DATA_STREAM_DATASET_DEFAULT = null;
+
+  public static final String DATA_STREAM_TYPE_CONFIG = "data.stream.type";
+  private static final String DATA_STREAM_TYPE_DOC =
+      "Generic type describing the data to be written to data stream. "
+      + "Should only be ``" + DataStreamType.LOGS.name() + "`` or ``"
+      + DataStreamType.METRICS.name() + "`` if the user wants to write "
+      + "to data stream. Otherwise, the default is" + DataStreamType.NONE.name()
+      + " which indicates the connector will write to regular indices instead. "
+      + "If set, this configuration will be used alongside " + DATA_STREAM_DATASET_CONFIG
+      + " to construct the index name.";
```

Comment:
```suggestion
  private static final String DATA_STREAM_TYPE_DOC =
      "Generic type describing the data to be written to data stream. "
      + "The default is" + DataStreamType.NONE.name()
      + " which indicates the connector will write to regular indices instead. "
      + "If set, this configuration will be used alongside " + DATA_STREAM_DATASET_CONFIG
      + " to construct the index name in the form {" + DATA_STREAM_TYPE_CONFIG +"}-{}-{} (and etc)";
```

use string.format() here